sigstore/_internal/_utils.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2022 The Sigstore Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from cryptography.x509 import Certificate, PrecertificateSignedCertificateTimestamps
+from cryptography.x509.certificate_transparency import SignedCertificateTimestamp
+
+
+class UnexpectedSctCountException(Exception):
+    """
+    Number of percerts scts is wrong
+    """
+
+    pass
+
+
+def get_sct_from_certificate(certificate: Certificate) -> SignedCertificateTimestamp:
+    # Try to retrieve the embedded SCTs within the cert.
+    precert_scts_extension = certificate.extensions.get_extension_for_class(
+        PrecertificateSignedCertificateTimestamps
+    ).value
```

Comment:
Now that we're running this code on certs coming from even more unknown sources, I think we need to add validation: It's likely just ExtensionNotFound that needs to be handled -- maybe turn it into ValueError here so the calling sites can just handle ValueError if they want?